### PR TITLE
[#41] [Android] [iOS] Setup project app flavors

### LINKED ISSUE #41: [Android] [iOS] Setup project app flavors
## Why

We need to prepare 2 app build flavors: staging and production for each Android and iOS app. These app build flavors help us to config and distribute corresponding apps on each staging & production API environment.

## Acceptance Criteria

- Setup project app flavors: `staging` and `production`.
- Setup app names.
- Setup app icons.

### PR DESCRIPTION
**Pull request by luongvo**
- Close #41

## What happened 👀

- Setup project app flavors: `staging` and `production` by using [BuildKonfig](https://github.com/yshrsmz/BuildKonfig).

## Insight 📝

- I followed this guide to set up iOS app flavors https://medium.com/macoclock/xcode-schemes-to-manage-staging-production-environment-in-swift-and-objective-c-e7be028c3141.
  - App names:
    - `KMM Survey - Staging` for staging apps.
    - `KMM Survey` for production apps.
  - For app icons: I used this tool https://easyappicon.com to generate app icons for each Android and iOS app, staging and production env.

## Proof Of Work 📹

- Android

- iOS

### CODE REVIEW

File: buildSrc/src/main/java/Configurations.kt [hunk 1]
```diff
@@ -1,3 +1,14 @@
+object Flavors {
+    const val PRODUCTION = "production"
+    const val STAGING = "staging"
+    const val DIMENSION = "stage"
```
Comment (blyscuit) [inline buildSrc/src/main/java/Configurations.kt]: What is the Dimension flavor? should its value be `"dimension"`?

**Review by blyscuit — APPROVED** [review]

File: buildSrc/src/main/java/FileExt.kt [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import java.io.File
+import java.util.*
+
+fun File.loadGradleProperties(fileName: String): Properties {
```
Comment (manh-t) [inline buildSrc/src/main/java/FileExt.kt:4]: should we create a directory `extensions` and put this file into it?

Comment (luongvo) [reply buildSrc/src/main/java/FileExt.kt:4]: @manh-t I think we don't need to move this into a new extensions package. 
- I don't see many people do that
  - https://github.com/gradle/kotlin-dsl-samples/blob/master/samples/multi-kotlin-project-with-buildSrc/buildSrc/src/main/kotlin/utils.kt
  - https://github.com/gradle/kotlin-dsl-samples/tree/master/samples/project-with-buildSrc/buildSrc/src/main/kotlin
- IMO, this is a simple buildSrc module so keeping all classes at the root package is still acceptable. Moving it into a new package produces more required `import` added.

image


As this IC focuses more on KMM, please allow me to ignore this issue 🙏

Comment (manh-t) [reply buildSrc/src/main/java/FileExt.kt:4]: @luongvo okay then

**Review by manh-t — APPROVED** [review]

File: buildSrc/src/main/java/Configurations.kt (hunk 1 above)
Comment (luongvo) [reply buildSrc/src/main/java/Configurations.kt]: @blyscuit It's a required [build-variant](https://developer.android.com/studio/build/build-variants#product-flavors) config. I changed it to `DIMENSION_VERSION = "version"` https://github.com/luongvo/kmm-survey/pull/42/commits/4e2294990cdd0be98308ad223e1ac68491145a44.

**Pull request merged**